syz-verifier/stats/stats_test.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2021 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package stats
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func getTestStats() *Stats {
+	return &Stats{
+		Runs:       20,
+		Mismatches: 10,
+		Calls: map[string]int{
+			"foo": 3,
+			"bar": 2,
+			"tar": 5,
+			"biz": 0,
+		},
+	}
+}
+
+func TestReportCallStats(t *testing.T) {
+	tests := []struct {
+		name, call, report string
+	}{
+		{
+			name:   "report for unsupported call",
+			call:   "read",
+			report: "",
+		},
+		{
+			name: "report for supported call",
+			call: "foo",
+			report: "mismatches for call foo:\n" +
+				"\t↳ out of total runs: 3 / 20 (15.00 %)\n" +
+				"\t↳ out of total mismatches: 3 / 10 (30.00 %)\n",
+		},
+	}
+	for _, test := range tests {
+		s := getTestStats()
+		t.Run(test.name, func(t *testing.T) {
+			got, want := s.ReportCallStats(test.call), test.report
+			if diff := cmp.Diff(got, want); diff != "" {
+				t.Errorf("s.ReportCallStats mismatch (-want +got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestReportGlobalStats(t *testing.T) {
+	s := getTestStats()
+	out := bytes.Buffer{}
+	s.ReportGlobalStats(&out)
+	got, want := out.String(),
+		"Statistics for run with 20 verifications:\n"+
+			"\t↳ mismatches: 10 / 20 (50.00 %)\n\n"+
+			"mismatches for call tar:\n"+
+			"\t↳ out of total runs: 5 / 20 (25.00 %)\n"+
+			"\t↳ out of total mismatches: 5 / 10 (50.00 %)\n\n"+
```

Comment:
What if we get more than 1 mismatch in a single program? Will the percentages sum up to >100% ?